client/client.go

```diff
@@ -1,44 +1,135 @@
 package client
 
 import (
+	"bytes"
 	"context"
-	"fmt"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
 
 	"github.com/CircleCI-Public/circleci-cli/logger"
 	"github.com/CircleCI-Public/circleci-cli/version"
-	"github.com/circleci/graphql"
+	"github.com/pkg/errors"
 )
 
-// NewClient returns a reference to a Client.
-// We also call graphql.NewClient to initialize a new GraphQL Client.
-// Then we pass the Logger originally constructed as cmd.Logger.
-func NewClient(endpoint string, logger *logger.Logger) *graphql.Client {
+// A Client is an HTTP client for our GraphQL endpoint.
+type Client struct {
+	endpoint   string
+	httpClient *http.Client
+	logger     *logger.Logger
+}
 
-	client := graphql.NewClient(endpoint)
+// RequestBody is used for serializing a request made up of a query and key/value pair of variables to send to the GraphQL server.
+type RequestBody struct {
```

Comment:
What do you think about merging this struct with the `Request` struct below and adding the json serialization to that?